src/main/java/seedu/address/logic/commands/ListCommand.java

```diff
@@ -5,30 +5,155 @@
 import seedu.address.model.Model;
 import seedu.address.model.pair.Pair;
 import seedu.address.model.person.Elderly;
 import seedu.address.model.person.Volunteer;
 
 /**
- * Lists all elderly, volunteers and pairs in FriendlyLink.
+ * Lists all elderly, volunteers and pairs in FriendlyLink based on the action given.
  */
 public class ListCommand extends Command {
 
     public static final String COMMAND_WORD = "list";
     public static final HashMap<Prefix, String> COMMAND_PROMPTS = new LinkedHashMap<>();
-    public static final String MESSAGE_SUCCESS = "Listed all elderly, volunteers and pairs";
+    public static final String MESSAGE_SUCCESS = "Listed all volunteers, elderly and pairs";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": List all volunteers and elderly pairs if nothing is"
+            + "specified after the command, else list all paired or unpaired volunteers and elderly if the"
+            + "\"paired\" or \"unpair\" (case-insensitive) word is given respectively after the the list command word"
+            + "Parameters: [paired/unpair]\n"
+            + "Example: " + COMMAND_WORD + " paired";
+    public static final String MESSAGE_SUCCESS_LIST_PAIRED =
+            "Listed %1$s paired volunteer, %2$s paired elderly and all pairs";
+    public static final String MESSAGE_SUCCESS_LIST_UNPAIRED =
+            "Listed %1$s unpaired volunteer, %2$s unpaired elderly and all pairs";
+
+    private final String action;
+
+    /**
+     * Constructs a ListCommand to display the lists of elderly, volunteers and pairs.
+     *
+     * @param trimmedArgs action of the {@code ListCommand}.
+     */
+    public ListCommand(String trimmedArgs) {
+        this.action = trimmedArgs;
+    }
 
     @Override
-    @SuppressWarnings("unchecked")
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        model.updateFilteredElderlyList((Predicate<Elderly>) PREDICATE_SHOW_ALL);
-        model.updateFilteredVolunteerList((Predicate<Volunteer>) PREDICATE_SHOW_ALL);
-        model.updateFilteredPairList((Predicate<Pair>) PREDICATE_SHOW_ALL);
-        return new CommandResult(MESSAGE_SUCCESS);
+        FriendlyLink friendlyLink = model.getFriendlyLink();
+
+        @SuppressWarnings("unchecked")
+        Predicate<Pair> showAllPairPredicate = (Predicate<Pair>) PREDICATE_SHOW_ALL;
+
+        switch (action) {
+        case "":
+            model.refreshAllFilteredLists();
+            return new CommandResult(MESSAGE_SUCCESS);
+        case "unpaired":
+            model.updateFilteredElderlyList(getUnPairedElderlyPredicate(friendlyLink));
+            model.updateFilteredVolunteerList(getUnPairedVolunteerPredicate(friendlyLink));
+            model.updateFilteredPairList(showAllPairPredicate);
+            return new CommandResult(String.format(MESSAGE_SUCCESS_LIST_UNPAIRED,
+                    model.getFilteredVolunteerList().size(), model.getFilteredElderlyList().size()));
```

Comment:
```suggestion
            model.refreshAllFilteredList(
                getUnPairedElderlyPredicate(friendlyLink),
                getUnPairedVolunteerPredicate(friendlyLink), 
                showAllPairPredicate);
            return new CommandResult(String.format(MESSAGE_SUCCESS_LIST_UNPAIRED,
                    model.getFilteredVolunteerList().size(), model.getFilteredElderlyList().size()));
```
Can you overload `refreshAllFilteredLists`function to take in these predicates instead?

Reply:
Every predicate is different, the code will be worse.